llvm/tools/llvm-dwarfdump/llvm-dwarfdump.cpp

```diff
@@ -286,6 +286,8 @@ static opt<bool> Verify("verify", desc("Verify the DWARF debug info."),
                         cat(DwarfDumpCategory));
 static opt<ErrorDetailLevel> ErrorDetails(
     "error-display", init(Unspecified),
+    desc("Set the level of detail and summary to display when verifying "
+         "(implies --verify)"),
     values(clEnumValN(NoDetailsOrSummary, "quiet",
                       "Only display whether errors occurred."),
            clEnumValN(NoDetailsOnlySummary, "summary",
```

Comment:
how about we just add a new `--error-detail` type named `json-summary`? The `summary` will emit text and the `json-summary` will emit the summary as json. This avoids the new and extra option and how to name it. Thoughts?

Reply:
That's another "axis" for the enumeration for [--error-display](url), unless you'd define it to be that if you're asking for JSON output, you just should get details in the JOSN and no details or summary to stdout. It's not a terrible idea, but I'm not well informed about the variety of ways folks currently use `--verify`.

Reply:
Throwing another idea out there: `--output-format=<enum>` where `<enum>` can be `json` or `text`? Then we document saying this applies to `--verify` only for now and can adjust that message if other options adopt this setting? Then we would send the JSON to STDOUT, which is similar to how `--statistics` does it?

Reply:
The challenge here is that right now we are only printing the summary (i.e. count + error category) to the json file, so it's not really about the "output-format" of everything printed by the verifier... We should probably let this idea bake in the oven for longer and for now just merge this with a rename of the flag added in the patch?

Reply:
Output for this "class" of tool (gsymutil as well) probably ought to be structured to better support tooling validation. This is step one. I'm happy to make a shorter alias for `--json-error-summary-file` (`-j` ?). I do think that the file format @clayborg suggested (and I adopted) can be extended to easily include full details for each error.

Reply:
Maybe `--json-verification-summary` to make it clear that the verification is implied, even if you're just looking at the command line?

Reply:
`--verify-json` or `--verify-json-file`? Simpler and shorter?

Reply:
> `--verify-json` or `--verify-json-file`? Simpler and shorter?

That would also sort it in with the `--verify` help better. Just pushed the update.